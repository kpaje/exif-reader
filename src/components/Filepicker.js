import React from "react";

function Filepicker(props) {
  return (
    <div>
      <input id="filepicker" onChange={props.onChange} type="file"></input>
    </div>
  );
}

export default Filepicker;
