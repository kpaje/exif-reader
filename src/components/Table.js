import React from "react";

function Table({ tableData }) {
	function formatTable(tableData) {
		let result = Object.entries(tableData).map(([key, value]) => (
			<tr key={key}>
				<td style={styles.table}>{key}</td>
				<td style={styles.table}>{value.toString()}</td>
			</tr>
		));
		return result;
	}

	return (
		<table style={styles.table}>
			{!undefined || !null ? <tbody>{formatTable(tableData)}</tbody> : null}
		</table>
	);
}

export default Table;

const styles = {
	table: {
		textAlign: "left",
		border: "1px solid",
		borderCollapse: "collapse"
	}
};
