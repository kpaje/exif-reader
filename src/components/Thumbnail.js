import React from "react";

function Thumbnail(thumbnail) {
	let imgURL = null;
	//create blob from image to link url for thumbnail
	if (thumbnail.src) {
		imgURL = URL.createObjectURL(thumbnail.src);
	}
	return (
		<div>
			{imgURL !== null ? (
				<img src={imgURL} alt={thumbnail.alt} className="App-logo" />
			) : null}
		</div>
	);
}

export default Thumbnail;
