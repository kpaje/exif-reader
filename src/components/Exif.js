import React, { useState } from "react";
import "../App.css";
import * as exifr from "exifr";
import { formatObj } from "../utils/helpers";
import { definedProps, detectThumbnail } from "../api/exif-js";
import Thumbnail from "./Thumbnail";
import Filepicker from "./Filepicker";
import Table from "./Table";
import GoogleMap from "../api/googleMap";

function Exif() {
	const [thumbnail, setThumbnail] = useState();
	const [tableData, setTableData] = useState({});
	const [lat, setLat] = useState("");
	const [lng, setLng] = useState("");

	const handlers = {
		getTableData: function(exifData) {
			setTableData(exifData);
		},
		getThumbnail: function(image) {
			let thumbnail = detectThumbnail(image);
			setThumbnail(thumbnail);
		},
		getGPSCoordinates: function(exif) {
			setLat(exif.latitude);
			setLng(exif.longitude);
		}
	};

	const processExif = (exif, image) => {
		let exifData = formatObj(exif, definedProps);
		handlers.getGPSCoordinates(exifData);
		handlers.getTableData(exifData);
		handlers.getThumbnail(image);
	};

	const generateExif = event => {
		const image = event.target.files[0];
		exifr
			.parse(image)
			.then(exif => {
				processExif(exif, image);
			})
			.catch(console.error);
	};

	return (
		<div>
			<header className="App-header">
				<Thumbnail src={thumbnail} alt={thumbnail} />
				<Filepicker onChange={generateExif} />
				<Table tableData={tableData} />
			</header>
			<GoogleMap lat={lat} lng={lng} />
		</div>
	);
}

export default Exif;
