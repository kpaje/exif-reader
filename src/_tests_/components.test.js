import React from "react";
import Exif from "../components/Exif";
import Filepicker from "../components/Filepicker";
import FingerprintTable from "../components/FingerprintTable";
import Table from "../components/Table";
import Thumbnail from "../components/Thumbnail";
import renderer from "react-test-renderer";

describe("Component Rendering", () => {
	it("My Test Case", () => {
		expect(true).toEqual(true);
	});

	it("Exif Component", () => {
		const ExifComponent = renderer.create(<Exif />).toJSON();
		expect(ExifComponent).toMatchSnapshot();
	});

	it("FingerprintTable Component", () => {
		const FingerprintTableComponent = renderer
			.create(<FingerprintTable />)
			.toJSON();
		expect(FingerprintTableComponent).toMatchSnapshot();
	});

	it("Table Component", () => {
		const TableComponent = renderer.create(<Table />).toJSON();
		expect(TableComponent).toMatchSnapshot();
	});

	it("Thumbnail Component", () => {
		const ThumbnailComponent = renderer.create(<Thumbnail />).toJSON();
		expect(ThumbnailComponent).toMatchSnapshot();
	});
});
