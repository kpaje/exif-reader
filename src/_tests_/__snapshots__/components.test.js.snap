// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Rendering Exif Component 1`] = `
<div>
  <header
    className="App-header"
  >
    <div />
    <div>
      <input
        id="filepicker"
        onChange={[Function]}
        type="file"
      />
    </div>
    <table
      style={
        Object {
          "border": "1px solid",
          "borderCollapse": "collapse",
          "textAlign": "left",
        }
      }
    >
      <tbody />
    </table>
  </header>
  <div>
    Loading...
  </div>
</div>
`;

exports[`Component Rendering Thumbnail Component 1`] = `<div />`;
